### ISSUE TITLE
Unable to create wheel using pip

### ISSUE THREAD

**Issue by cyraxjoe** [Changaco/setuptools_scm_git_archive#9]
Apparently `pip` can't create a wheel for this package, you end up with something like:
 `pip wheel --no-binary ":all:" setuptools_scm_git_archive`

```
$ pip wheel --no-binary ":all:" setuptools_scm_git_archive
Collecting setuptools_scm_git_archive
  Using cached https://files.pythonhosted.org/packages/18/00/754438f629dfff1ecdbb31c8c23effe8bcfdbf65dbaf1cbed58eab286d82/setuptools_scm_git_archive-1.0.tar.gz
Building wheels for collected packages: setuptools-scm-git-archive
  Building wheel for setuptools-scm-git-archive (setup.py) ... error
  Complete output from command /home/joe/envs/dev/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-wheel-c9k89q3b/setuptools-scm-git-archive/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-jqxaac2f:
  running bdist_wheel
  The [wheel] section is deprecated. Use [bdist_wheel] instead.
  running build
  running build_py
  creating build
  creating build/lib
  creating build/lib/setuptools_scm_git_archive
  copying setuptools_scm_git_archive/__init__.py -> build/lib/setuptools_scm_git_archive
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/setuptools_scm_git_archive
  copying build/lib/setuptools_scm_git_archive/__init__.py -> build/bdist.linux-x86_64/wheel/setuptools_scm_git_archive
  running install_egg_info
  running egg_info
  creating setuptools_scm_git_archive.egg-info
  writing setuptools_scm_git_archive.egg-info/PKG-INFO
  writing dependency_links to setuptools_scm_git_archive.egg-info/dependency_links.txt
  writing entry points to setuptools_scm_git_archive.egg-info/entry_points.txt
  writing top-level names to setuptools_scm_git_archive.egg-info/top_level.txt
  writing manifest file 'setuptools_scm_git_archive.egg-info/SOURCES.txt'
  reading manifest file 'setuptools_scm_git_archive.egg-info/SOURCES.txt'
  writing manifest file 'setuptools_scm_git_archive.egg-info/SOURCES.txt'
  Copying setuptools_scm_git_archive.egg-info to build/bdist.linux-x86_64/wheel/setuptools_scm_git_archive-0-py3.6.egg-info
  running install_scripts
  adding license file "LICENSE" (matched pattern "LICEN[CS]E*")
  creating build/bdist.linux-x86_64/wheel/setuptools_scm_git_archive-0.dist-info/WHEEL
  creating '/tmp/pip-wheel-jqxaac2f/setuptools_scm_git_archive-0-py2.py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
  adding 'setuptools_scm_git_archive/__init__.py'
  adding 'setuptools_scm_git_archive-0.dist-info/LICENSE'
  adding 'setuptools_scm_git_archive-0.dist-info/METADATA'
  adding 'setuptools_scm_git_archive-0.dist-info/WHEEL'
  adding 'setuptools_scm_git_archive-0.dist-info/entry_points.txt'
  adding 'setuptools_scm_git_archive-0.dist-info/top_level.txt'
  adding 'setuptools_scm_git_archive-0.dist-info/RECORD'
  removing build/bdist.linux-x86_64/wheel
  running bdist_wheel
  The [wheel] section is deprecated. Use [bdist_wheel] instead.
  running build
  running build_py
  installing to build/bdist.linux-x86_64/wheel
  running install
  running install_lib
  copying build/lib/setuptools_scm_git_archive/__init__.py -> build/bdist.linux-x86_64/wheel/setuptools_scm_git_archive
  error: could not create 'build/bdist.linux-x86_64/wheel/setuptools_scm_git_archive/__init__.py': No such file or directory
 
```
I assume the error gets caused because `setuptools_scm_git_archive` is using _itself_ to setup the version right? So the wheel can only be created if you're working on a checkout not from a source distribution. If that's the case would adding the git metadata into the source distribution would be undesired?

The source distribution in my mind should be able to create different kind of binary distributions of itself... but I guess this package might be an exception to the rule.

Thoughts?

**Comment by cyraxjoe** [comment]
To add more information.

Downloading the source distribution and doing `python setup.py bdist_wheel` end up with the same result... the only working method to create the wheel seems to be:

  1. Clone the repo
  2. Checkout the `1.0` tag 
  3. python setup.py bdist_wheel

**Comment by cyraxjoe** [comment]
Apparently the problem is caused because of the double `setup` call, if I remove the first one and just add the local path to the working set and the final setup call, seems to work fine. Also, the[ source distribution in pypi ](https://files.pythonhosted.org/packages/18/00/754438f629dfff1ecdbb31c8c23effe8bcfdbf65dbaf1cbed58eab286d82/setuptools_scm_git_archive-1.0.tar.gz)doesn't have the same setup.py file. The one in pypi looks for a  `.git` directory not tries to use the entrypoint.

**Comment by cyraxjoe** [comment]
Fixed after margin PR #10. Thank you @Changaco
